reduce() was removed in Python 3

flake8 testing of https://github.com/thumbor/thumbor on Python 3.6.2

$ __flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics__
```
./thumbor/error_handlers/file.py:44:27: F821 undefined name 'reduce'
                    obj = reduce(getattr, parts[1:], obj)
                          ^
```
https://github.com/thumbor/thumbor/blob/master/thumbor/error_handlers/file.py#L44

https://docs.python.org/3/whatsnew/3.0.html#builtins  Removed reduce(). Use `functools.reduce()` if you really need it; however, 99 percent of the time an explicit for loop is more readable.

--- a/thumbor/error_handlers/file.py
+++ b/thumbor/error_handlers/file.py
@@ -3,6 +3,7 @@
 
 import logging
 import logging.handlers
+import operator
 import json
 import re
 import time
@@ -24,7 +25,7 @@ class ErrorHandler(object):
             )
         if (config.ERROR_FILE_NAME_USE_CONTEXT and not re.search('^(\w+\.)?\w+$', config.ERROR_FILE_NAME_USE_CONTEXT)):
             raise RuntimeError(
-                "ERROR_FILE_NAME_USE_CONTEXT config must reffer an attribute of context "
+                "ERROR_FILE_NAME_USE_CONTEXT config must refer an attribute of context "
                 "object and be form of ^(\w+.)?\w+$ : %s" % config.ERROR_FILE_NAME_USE_CONTEXT
             )
         self.file_name = config.ERROR_FILE_LOGGER
@@ -38,12 +39,7 @@ class ErrorHandler(object):
         # create log file if not existing
         if not self.logger:
             if self.use_context:
-                if '.' in self.use_context:
-                    parts = self.use_context.split('.')
-                    obj = getattr(context, parts[0], None)
-                    obj = reduce(getattr, parts[1:], obj)
-                else:
-                    obj = getattr(context, self.use_context, None)
+                obj = operator.attrgetter(self.use_context)(context)
                 file = self.file_name % obj
             else:
                 file = self.file_name

